gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/internal/PluginRegistryImpl.java

```diff
@@ -122,79 +128,79 @@ public void init() throws Exception {
             pluginsPath = workspacesPath;
         }
 
-        for (String aWorkspacePath : pluginsPath) {
-            loadPluginsFromRegistry(aWorkspacePath);
-        }
-
-        printPlugins();
-        eventManager.publishEvent(PluginEvent.ENDED, null);
+        Flowable
+            .fromArray(pluginsPath)
+            .flatMap(this::loadPluginsFromPath)
+            // reserve sort
+            .sorted((p1, p2) -> new Long(p2.getArchiveTimestamp() - p1.getArchiveTimestamp()).intValue())
+            .filter(candidate -> {
+                // As plugin arrive sorted by reverse file date
+                // we can exclude duplicate
+                return !plugins.stream().filter(registered -> registered.equals(candidate)).findFirst().isPresent();
+            })
+            .doOnNext(plugins::add)
+            .doOnNext(plugin -> eventManager.publishEvent(PluginEvent.DEPLOYED, plugin))
+            .doOnTerminate(() -> {
```

Comment:
Shouldn't be doOnComplete ?

Also shouldn't we add a onErrorResumeNext to deal with error?

Reply:
I've used doOnComplete which is more accurate indeed. That in combination with blockingAwait() does the trick of rethrowing runtime exceptions